deployment_report/model/utils/ingress_util.py

```diff
@@ -120,51 +120,56 @@ def get_ingress_version(kubectl: KubectlInterface) -> Optional[Text]:
     """
 
     version: Text = ""
-    getpod_cmd: AnyStr = "get pods -n " + kubectl.ingress_ns + \
-                         " --field-selector=status.phase==Running" + \
-                         " -o jsonpath=\"{.items[0].metadata.name}\""
-
-    if kubectl.ingress_ns == SupportedIngress.Controllers.NS_NGINX:
-        podname: AnyStr = kubectl.do(getpod_cmd +
-                                     " -l app.kubernetes.io/component=controller")
-        version_str: AnyStr = kubectl.do("exec -it " + podname.decode() +
-                                         " -n " + kubectl.ingress_ns +
-                                         " -- /nginx-ingress-controller --version")
-        version_list: List = version_str.decode().splitlines()
-        for v in version_list:
-            if _RELEASE_ in v:
-                version = ' '.join(v.split()) + version
-            elif _NGINX_VERSION_ in v:
-                version = version + ", " + v.split()[-1]
-
-    elif kubectl.ingress_ns == SupportedIngress.Controllers.NS_ISTIO:
-        podname: AnyStr = kubectl.do(getpod_cmd +
-                                     " -l  app=istiod")
-        version_str: AnyStr = kubectl.do("exec -it " + podname.decode() +
-                                         " -n " + kubectl.ingress_ns +
-                                         " -- pilot-discovery version --short")
-        version = version_str.decode()
-
-    elif kubectl.ingress_ns == SupportedIngress.Controllers.NS_OPENSHIFT:
-        podname: AnyStr = kubectl.do(getpod_cmd +
-                                     " -l  name=ingress-operator")
-        version_str: AnyStr = kubectl.do("get pod " + podname.decode() +
-                                         " -n " + kubectl.ingress_ns +
-                                         " -o jsonpath=\"{.spec.containers[].env[" +
-                                         "?(@.name=='RELEASE_VERSION')].value}\"")
-        version = version_str.decode()
-
-    elif kubectl.ingress_ns == SupportedIngress.Controllers.NS_CONTOUR:
-        podname: AnyStr = kubectl.do(getpod_cmd +
-                                     " -l app=envoy")
-        version_str: AnyStr = kubectl.do("get pod " + podname.decode() +
-                                         " -n " + kubectl.ingress_ns +
-                                         " -o jsonpath=\"{.spec.containers[*].image}\"")
-        version_list: List = version_str.decode().split(' ')
-        for v in version_list:
-            if version:
-                version = version + ", " + v.split("/")[-1].capitalize()
-            else:
-                version = v.split("/")[-1].capitalize()
-
-    return version.strip()
+
+    try:
```

Comment:
the use of try catch over the entire section may be excessive, instead have it check to make sure the variables being accessed are present and if not then it can return in an appropriate way.

Reply:
Variables were checked before the function call. it seems there is no need to check a second time.

Reply:
The new general try/catch over the entire function should be removed since that could cause other exceptions to be handled where it shouldn't perhaps

Reply:
Mmm... can you detail what other exceptions in this function need to be handled?

Reply:
Removed the try/catch, add podname check.